### ISSUE TITLE
Adding a public symbol to with-magling makes it private

### ISSUE THREAD

**Issue by xhochy** [jemalloc/jemalloc#1855]
If you mangle e.g. `aligned_alloc` using `--with-mangling=aligned_alloc:__aligned_alloc`, it will not turn up as `__aligned_alloc` in the generated shared object but will have the private namespace prefix (in my case it is named `je___aligned_alloc`).

The offensive code is in https://github.com/jemalloc/jemalloc/blob/8da0896b7913470250a0220504822028e2aa8f2a/configure.ac#L2282 where the symbol is removed but not added again with the new name.

**Comment by xhochy** [comment]
I solved this on conda-forge using the following patch: https://github.com/conda-forge/jemalloc-feedstock/blob/61bfe05bedc989f1978abc5b43db212886263c1b/recipe/export-mangled-syms.patch

This probably only works as long as all symbols in `--with-mangling` are public symbols.

**Comment by davidtgoldblatt** [comment]
Sorry for the long delay on this -- I'm working on a big build system change in the background and didn't want to hit a moving target; that change has taken longer than expected.

If keeping the patch out of upstream is particularly painful for you we can make this sort of change sooner though; it's not that hard to adapt to.

**Comment by xhochy** [comment]
No, for me it is fine to keep this patch only in the conda recipe if it is not easy to integrate here.
